### DISCUSSION TITLE
Reduce the Dependencies

### DISCUSSION THREAD

**Discussion by timbomckay** [tailwindlabs/tailwindcss#12845] [Ideas]
I use Tailwind with a Hugo static site. It's the only NPM dependency I have and the rest is all powered through Hugo's pipelines. Unfortunately, while I only have a dependency on Tailwind, it means I also have a dependency on Tailwind's dependencies. Which is currently 22, but with the nested dependencies it ends up with 112 packages ([according to pkg-size](https://pkg-size.dev/tailwindcss)).

I love Tailwind and this won't be a swaying factor, but one of the reasons I strive for this minimalism is to reduce the likelihood for dependency vulnerabilities. The odds are low with this dependency tree, but I have seen it with various `glob` & `postcss` dependencies.

I know there's the [CDN option](https://tailwindcss.com/docs/installation/play-cdn), but since that's not ideal for production perhaps something like that asn NPM package could be available as a minimal, bundled option.

**Comment by adamwathan** [comment]
We're currently rewriting Tailwind for v4 and you'll be happy to know that right now we only have one dependency, which is `lightningcss` 👍  Striving to keep it as light as possible this time around.
